engine/types/type42/linking_key_test.go

```diff
@@ -0,0 +1,149 @@
+package type42
+
+import (
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"github.com/libsv/go-bk/bec"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCalculateHMAC(t *testing.T) {
+	tests := []struct {
+		name            string
+		pubSharedSecret []byte
+		reference       string
+		idx             int
+		expected        []byte
+		wantErr         bool
+	}{
+		{
+			name:            "valid input",
+			pubSharedSecret: []byte("shared_secret"),
+			reference:       "reference",
+			idx:             1,
+			expected:        computeExpectedHMAC([]byte("shared_secret"), "reference", 1),
```

Comment:
If we really want to test `calculateHMAC`, then we should have those expected values as static string, otherwise we can introduce bug in computeExpectedHMAC and get false positives or false negatives from those tests.